script/deploy/RelayDeploy.sol

```diff
@@ -0,0 +1,192 @@
+     * @return The address of the deployed KeyRegistry contract
+     */
+    function deployKeyRegistry(address proxyOwner, bool isDeployerGuarded) public virtual returns (address) {
+        vm.startBroadcast();
+        (address implementation, bytes memory initData) = _keyRegistryParams();
+        address newContract =
+            _deployContract(KEY_REGISTRY_SALT, implementation, initData, proxyOwner, isDeployerGuarded);
+        vm.stopBroadcast();
+        Logs.log(string.concat("KeyRegistry deployed at: ", vm.toString(newContract)));
+
+        return newContract;
+    }
+
+    /**
+     * @notice Deploy the ValSetDriver contract using CREATE3
+     * @dev Deploys a transparent upgradeable proxy for the ValSetDriver
+     * @param proxyOwner The owner of the proxy contract
+     * @param isDeployerGuarded Whether to deploy with guarded salt for enhanced security
+     * @return The address of the deployed ValSetDriver contract
+     */
+    function deployDriver(address proxyOwner, bool isDeployerGuarded) public virtual returns (address) {
+        vm.startBroadcast();
+        (address implementation, bytes memory initData) = _driverParams();
+        address newContract = _deployContract(DRIVER_SALT, implementation, initData, proxyOwner, isDeployerGuarded);
+        vm.stopBroadcast();
+        Logs.log(string.concat("ValSetDriver deployed at: ", vm.toString(newContract)));
+
+        return newContract;
+    }
+
+    /**
+     * @notice Deploy the Settlement contract using CREATE3
+     * @dev Deploys a transparent upgradeable proxy for the Settlement
+     * @param proxyOwner The owner of the proxy contract
+     * @param isDeployerGuarded Whether to deploy with guarded salt for enhanced security
+     * @return The address of the deployed Settlement contract
+     */
+    function deploySettlement(address proxyOwner, bool isDeployerGuarded) public virtual returns (address) {
+        vm.startBroadcast();
+        (address implementation, bytes memory initData) = _settlementParams();
+        address newContract = _deployContract(SETTLEMENT_SALT, implementation, initData, proxyOwner, isDeployerGuarded);
+        vm.stopBroadcast();
+        Logs.log(string.concat("Settlement deployed at: ", vm.toString(newContract)));
+
+        return newContract;
+    }
+
+    /**
+     * @notice Internal function to deploy a contract using CREATE3 with optional initialization
+     * @dev Creates a transparent upgradeable proxy and optionally initializes it
+     * @param salt The CREATE3 salt for deterministic deployment
+     * @param implementation The implementation contract address
+     * @param initData The initialization data for the proxy (empty bytes if no initialization)
+     * @param owner The owner of the proxy contract
+     * @param isDeployerGuarded Whether to use guarded salt deployment
+     * @return The address of the deployed contract
+     */
+    function _deployContract(
+        bytes32 salt,
```

Comment:
I think it would be better to get bytes11 salt, and not truncate it inside the function